### PR TITLE
{ASI} :- Image size is not passed to image-uploader when inserting an image from new media gallery

### PR DESCRIPTION
**Pull request by konarshankar07**
### Description (*)

This PR is the part of Adobe stock integration

### Dependent to
https://github.com/magento/adobe-stock-integration/pull/1021

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#1010: Image size is not passed to image-uploader when inserting an image from new media gallery

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/MediaGalleryApi/Api/Data/AssetInterface.php [hunk 1]
```diff
@@ -65,6 +65,13 @@ public function getHeight(): int;
      */
     public function getWidth(): int;
 
+    /**
+     * Retrieve full licensed asset's size
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryApi/Api/Data/AssetInterface.php]:
```suggestion
     * Retrieve asset file size in bytes
```

File: app/code/Magento/MediaGallery/etc/db_schema.xml [hunk 2]
```diff
@@ -14,6 +14,7 @@
         <column xsi:type="varchar" name="content_type" length="255" nullable="true" comment="Content Type"/>
         <column xsi:type="int" name="width" padding="10" unsigned="true" nullable="false" identity="false" default="0" comment="Width"/>
         <column xsi:type="int" name="height" padding="10" unsigned="true" nullable="false" identity="false" default="0" comment="Height"/>
+        <column xsi:type="int" name="size" padding="10" unsigned="true" nullable="false" identity="false" default="0" comment="Size"/>
```
Comment (sivaschenko) [inline app/code/Magento/MediaGallery/etc/db_schema.xml]:
```suggestion
        <column xsi:type="int" name="size" padding="10" unsigned="true" nullable="false" identity="false" comment="Asset file size in bytes"/>
```

File: app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php [hunk 3]
```diff
@@ -54,22 +53,41 @@ public function testExtractData(string $class, $interfaceClass, array $expectedD
...
+            $this->checkKeyWordValues($expectedData, $receivedData, $model);
+        }
+    }
+
+    /**
+     * @param array $expectedData
+     * @param array $data
+     * @param object $model
+     */
+    protected function checkAssetValues(array $expectedData, array $data, $model)
```
Comment (rogyar) [inline app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php]: I would suggest using `private` scope for the private methods. `protected` scope assumes that we consider an inheritance of this test class by another test class. It's not a big deal, however, it's recommended to be strict with scoping and hide implementation details in such cases.

File: app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php [hunk 4]
```diff
@@ -122,7 +144,7 @@ public function assetProvider()
             ],
             'Keyword conversion without interface' => [
                 Keyword::class,
-                null,
+                '',
```
Comment (sivaschenko) [inline app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php:147]: this value should be either a valid interface name or null

File: app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php [hunk 5]
```diff
@@ -54,22 +53,41 @@ public function testExtractData(string $class, $interfaceClass, array $expectedD
                 'data' => $data,
             ]
         );
-        $receivedData = $this->dataExtractor->extract($model, $interfaceClass);
-        $this->checkValues($expectedData, $receivedData, $model);
+        if ($interfaceClass) {
```
Comment (sivaschenko) [inline app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php:56]: I don't see the reason to change the test logic after adding a `size` field. Only data provider should be changed. What is the reason for this change?

**Comment by rogyar** [comment]
@magento run all tests

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @konarshankar07 ! Please see my code review notes

File: app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php (hunk 5 above)
Comment (konarshankar07) [reply app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php:56]: I'm trying to improve the Unit test for this line https://github.com/magento/magento2/blob/1a77b70fc59ba3d0d1cae76f40be567baa75041e/app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php#L72

File: app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php (hunk 4 above)
Comment (konarshankar07) [reply app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php:147]: As per function argument type declarations for the method is string so instead of null I just replaced with the empty string

Comment (sivaschenko) [reply app/code/Magento/MediaGallery/Test/Unit/Model/DataExtractorTest.php:147]: Let's keep it logically correct: it should be either interface name as string or null

**Review by sivaschenko — APPROVED** [review]

**Comment by sivaschenko** [comment]
Failed tests look to be random and not relevant to the changes in the PR

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-7196](https://jira.corp.magento.com/browse/ENGCOM-7196) has been created to process this Pull Request
:eight_spoked_asterisk: @sivaschenko, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

Switched to the related Adobe Stock Integration PR [1021](https://github.com/magento/adobe-stock-integration/pull/1021) and to the current PR

The image size is displayed, no _NaN undefined_ error
27388_image size

**Comment by engcom-Echo** [comment]
Failed functional tests not related to the changes in this PR

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-7196](https://jira.corp.magento.com/browse/ENGCOM-7196) has been created to process this Pull Request
